ddl/ddl_api.go

```diff
@@ -806,6 +806,16 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 			buf := new(bytes.Buffer)
 			s.Partition.Expr.Format(buf)
 			pi.Expr = buf.String()
+			if s.Partition.Tp == model.PartitionTypeRange {
+				if _, ok := s.Partition.Expr.(*ast.ColumnNameExpr); ok {
+					for _, col := range cols {
+						name := strings.Replace(col.Name.String(), ".", "`.`", -1)
+						if fmt.Sprintf("`%s`", name) == pi.Expr && col.Tp != mysql.TypeLong {
```

Comment:
Although it doesn't matter here, I'd like to point out that this is better for performance:
```if col.Tp != mysql.TypeLong && fmt.Sprintf("`%s`", name) == pi.Expr {```

Reply:
DONE